mars/oscar/backends/ray/pool.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 1999-2020 Alibaba Group Holding Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import logging
+import os
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import List, Optional
+
+from .communication import ChannelID, RayServer
+from .utils import process_address_to_placement, process_placement_to_address, get_placement_group
+from ..config import ActorPoolConfig
+from ..pool import AbstractActorPool, MainActorPoolBase, SubActorPoolBase, create_actor_pool, _register_message_handler
+from ....utils import lazy_import
+
+ray = lazy_import('ray')
+logger = logging.getLogger(__name__)
+
+
+class RayActorPoolMixin(AbstractActorPool, ABC):
+
+    async def __on_ray_recv__(self, channel_id: ChannelID, message):
+        """Method for communication based on ray actors"""
+        if not hasattr(self, '_external_servers'):
+            ray_servers = [server for server in self._servers if isinstance(server, RayServer)]
```

Comment:
Why not set the `_external_servers` in `__init__`?

Reply:
ActorPool constructor has too many arguments, it will make RayMainActorPool/RaySUbActorPool constructor compilicated and doesn't get too much gains

Reply:
fixed